### PR TITLE
check more older R versions

### PR DESCRIPTION
**Pull request by IndrajeetPatil**

### CODE REVIEW

File: .github/workflows/R-check.yaml [hunk 1]
```diff
@@ -51,7 +57,10 @@ jobs:
 
       - uses: r-lib/actions/setup-r-dependencies@v2
         with:
-          extra-packages: any::rcmdcheck
+          extra-packages: |
+            any::rcmdcheck
+            lme4=?ignore-before-r=4.0.0
+            emmeans=?ignore-before-r=3.6.0
```
Comment (mattansb) [inline .github/workflows/R-check.yaml:63]: What do these do?

Comment (IndrajeetPatil) [reply .github/workflows/R-check.yaml:63]: It skips downloading these before a certain R version because they are not available for those R versions.

**Comment by IndrajeetPatil** [comment]
Why are you relying on the entire _tidymodels_ ecosystem? That's almost 100 packages to download.

``` r
tools::package_dependencies("tidymodels", recursive = TRUE)
#> $tidymodels
#>  [1] "broom"        "cli"          "conflicted"   "dials"        "dplyr"       
#>  [6] "ggplot2"      "hardhat"      "infer"        "modeldata"    "parsnip"     
#> [11] "purrr"        "recipes"      "rlang"        "rsample"      "rstudioapi"  
#> [16] "tibble"       "tidyr"        "tune"         "workflows"    "workflowsets"
#> [21] "yardstick"    "backports"    "ellipsis"     "generics"     "glue"        
#> [26] "methods"      "stringr"      "utils"        "memoise"      "scales"      
#> [31] "DiceDesign"   "lifecycle"    "vctrs"        "withr"        "magrittr"    
#> [36] "R6"           "tidyselect"   "pillar"       "digest"       "grDevices"   
#> [41] "grid"         "gtable"       "isoband"      "MASS"         "mgcv"        
#> [46] "stats"        "patchwork"    "globals"      "prettyunits"  "gower"       
#> [51] "ipred"        "lubridate"    "Matrix"       "timeDate"     "furrr"       
#> [56] "slider"       "fansi"        "pkgconfig"    "cpp11"        "foreach"     
#> [61] "GPfit"        "codetools"    "iterators"    "future"       "lhs"         
#> [66] "lattice"      "rpart"        "survival"     "nnet"         "class"       
#> [71] "prodlim"      "graphics"     "cachem"       "nlme"         "splines"     
#> [76] "crayon"       "utf8"         "farver"       "labeling"     "munsell"     
#> [81] "RColorBrewer" "viridisLite"  "warp"         "stringi"      "fastmap"     
#> [86] "listenv"      "parallel"     "parallelly"   "tools"        "Rcpp"        
#> [91] "colorspace"   "KernSmooth"   "lava"         "future.apply" "progressr"   
#> [96] "numDeriv"     "SQUAREM"
```

<sup>Created on 2022-06-12 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by IndrajeetPatil** [comment]
@mattansb Builds now fail with:

```r
❯ checking for unstated dependencies in ‘tests’ ... WARNING
  'library' or 'require' call not declared from: ‘tidymodels’
```

Even if you are skipping the tests, the CRAN static code analysis will still figure out that the tests rely on _tidymodels_. 
So you can't have the test _and_ not include the package in `Suggests` either.

**Comment by strengejacke** [comment]
Honestly, I have skipped tests for R 3.4 and 3.5 on GHA, to require fewer resources when testing. I think the most stuff is done that it works for R 3.4, so we don't need to test any more, I'd say. We should run tests every now and then with `_R_CHECK_FORCE_SUGGESTS_: true`, to test whether the package also works w/o suggested packages, but the backwards compatibility code should be included (like `isFALSE()` etc.) and work for R3.4/3.5

**Comment by IndrajeetPatil** [comment]
@mattansb There are a couple of tests that don't succeed on R < 3.6. I am assuming this has to do with the changes made to `set.seed()` in R 3.6. 

I think we should skip them for R versions < 3.6?

**Comment by mattansb** [comment]
@IndrajeetPatil I un-skipped those tests.
If that's the only thing failing, we can merge this PR (I'll fix those tests some other time).

**Pull request merged**